src/itwinai/loggers.py

```diff
@@ -240,7 +239,7 @@ def serialize(self, obj: Any, identifier: str) -> str:
         Returns:
             str: local path of the serialized object to be logged.
         """
-        itm_path = os.path.join(self.savedir, identifier)
+        itm_path = self.savedir / str(identifier)
```

Comment:
"identifier" should already be of type "str" here.

Reply:
true